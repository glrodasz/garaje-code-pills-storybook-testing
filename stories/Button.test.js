import * as React from "react";
import { render } from "@testing-library/react";
import { composeStories } from "@storybook/testing-react";

import defaultStory, * as stories from "./Button.stories";

const composedStories = composeStories(stories);

describe(`[ ${defaultStory.title} ]`, () => {
  Object.entries(composedStories).forEach(([story, Component]) => {
    it(`should render ${story}`, () => {
      const { asFragment } = render(<Component {...defaultStory.args} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
