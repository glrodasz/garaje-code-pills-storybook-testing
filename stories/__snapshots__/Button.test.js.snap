// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ Example/Button ] should render Large 1`] = `
<DocumentFragment>
  <button
    class="storybook-button storybook-button--large storybook-button--secondary"
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;

exports[`[ Example/Button ] should render Primary 1`] = `
<DocumentFragment>
  <button
    class="storybook-button storybook-button--medium storybook-button--primary"
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;

exports[`[ Example/Button ] should render Secondary 1`] = `
<DocumentFragment>
  <button
    class="storybook-button storybook-button--medium storybook-button--secondary"
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;

exports[`[ Example/Button ] should render Small 1`] = `
<DocumentFragment>
  <button
    class="storybook-button storybook-button--small storybook-button--secondary"
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;
